pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -898,7 +900,36 @@ func (p *Pod) ensureWorkloadOwnedByAllMembers(ctx context.Context, c client.Clie
 	return nil
 }
 
-func (p *Pod) ConstructComposableWorkload(ctx context.Context, c client.Client, r record.EventRecorder) (*kueue.Workload, error) {
+func (p *Pod) getWorkloadLabels(labelKeysToCopy []string) (map[string]string, error) {
+	if len(labelKeysToCopy) == 0 {
+		return nil, nil
+	}
+	podLabels := p.Object().GetLabels()
+	workloadLabels := make(map[string]string, len(labelKeysToCopy))
+	if !p.isGroup {
+		for _, labelKey := range labelKeysToCopy {
+			if labelValue, found := podLabels[labelKey]; found {
+				workloadLabels[labelKey] = labelValue
+			}
+		}
```

Comment:
This is completely the same logic as the generic reconciler:
So, could we commonize them?

```go
	jobLabels := job.Object().GetLabels()
	workloadLabels := make(map[string]string, len(r.labelKeysToCopy))
	for _, labelKey := range r.labelKeysToCopy {
		if labelValue, found := jobLabels[labelKey]; found {
			workloadLabels[labelKey] = labelValue
		}
	}
```

Reply:
Ok, good idea. So this is in fact filtering a map with a certain list of keys. I extracted this into `maps` library (added a generic `maps.FilterKeys` function and a bunch of unit tests). And I'm using it here and in the generic reconciler. Please take a look if it looks better.